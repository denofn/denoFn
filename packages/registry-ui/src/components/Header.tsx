import React from "react";
import { css } from "otion";

import { ArrowLeft } from "../icons/ArrowLeft";
import denoFn from "../icons/denofn_transparent.png";
import { ViewsContext } from "./Views";

export type HeaderProps = {
  title?: string;
};

export function Header({ title = "denoFn" }: HeaderProps) {
  const [{ current, manage }, dispatch] = React.useContext(ViewsContext);
  return (
    <header
      className={css({
        display: "flex",
        backgroundColor: "rgb(0, 0, 0)",
        color: "white",
        borderColor: "rgb(237,242,247)",
        borderBottomStyle: "solid",
        borderBottomWidth: "1px",
        height: "4rem",
      })}
    >
      <div
        className={css({
          display: "flex",
          alignItems: "flex-start",
          justifyContent: "space-between",
          marginLeft: "auto",
          marginRight: "auto",
          width: "100%",
          paddingLeft: "1.5rem",
          paddingRight: "1.5rem",
        })}
      >
        <div
          className={css({
            display: "flex",
          })}
        >
          {current === "manage" || current === "create" ? (
            <ArrowLeft
              onClick={() => dispatch({ current: "view-all" })}
              className={css({
                display: "inline-block",
                lineHeight: "1.5rem",
                marginTop: "1.25rem",
                marginBottom: "1.25rem",
                marginRight: "0.5rem",
                cursor: "pointer",
                fill: "white",
              })}
            />
          ) : (
            <img
              className={css({
                marginTop: ".5rem",
                height: "3rem",
                width: "3rem",
                lineHeight: "3rem",
                marginRight: ".5rem",
              })}
              src={denoFn}
              alt="denoFn logo"
            />
          )}
          <h4
            className={css({
              display: "inline-block",
              lineHeight: "1.5rem",
              marginTop: "1.25rem",
              marginBottom: "1.25rem",
            })}
          >
            {current === "manage" ? manage : current === "create" ? "create fn" : title}
          </h4>
        </div>
      </div>
    </header>
  );
}
