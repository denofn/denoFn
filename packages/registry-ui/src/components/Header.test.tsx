import { render, cleanup } from "@testing-library/react";
import React from "react";

import { Header } from "./Header";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});
