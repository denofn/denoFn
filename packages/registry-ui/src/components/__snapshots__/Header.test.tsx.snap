// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <header
    class="_zjik7 _14yn38p _1y5pc60 _pv9j5f _10mnup3 _m138fg _p13mol"
  >
    <div
      class="_zjik7 _5dh3r6 _2o6p8u _1o3nkn _1tx1l5v _1d3w5wq _ufxjcp _1wunid1"
    >
      <div
        class="_zjik7"
      >
        <img
          alt="denoFn logo"
          class="_r32fpw _cz8lfe _a88p1 _1i2fn2n _1gz641"
          src="denofn_transparent.png"
        />
        <h4
          class="_1r5gb7q _1dw7ik1 _2xh5x8 _9kwmmn _1tq8czj"
        >
          denoFn
        </h4>
      </div>
      <div
        class="_akdqam"
      >
        <button
          class="_1khn195 _1peqg37 _1y5pc60 _1oscket _1efi8gv _my9kfw _fxc8cu _zowvxu _luq9nz _e0dnmk _r24itg _rkiwex _807wit _16qj0cz _ryz482"
        >
          Sign in
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
